Gems/Terrain/Code/Tests/TerrainHeightGradientListTests.cpp

```diff
@@ -46,47 +43,74 @@ class TerrainHeightGradientListComponentTest : public ::testing::Test
         m_app.Destroy();
     }
 
-    void CreateEntity()
+    AZStd::unique_ptr<AZ::Entity> CreateEntity()
     {
-        m_entity = AZStd::make_unique<AZ::Entity>();
-        ASSERT_TRUE(m_entity);
-
-        // Create the required box component.
-        UnitTest::MockAxisAlignedBoxShapeComponent* boxComponent = m_entity->CreateComponent<UnitTest::MockAxisAlignedBoxShapeComponent>();
-        m_app.RegisterComponentDescriptor(boxComponent->CreateDescriptor());
+        auto entity = AZStd::make_unique<AZ::Entity>();
+        entity->Init();
+        return entity;
+    }
 
+    Terrain::TerrainHeightGradientListComponent* AddHeightGradientListToEntity(AZ::Entity* entity)
+    {
         // Create the TerrainHeightGradientListComponent with an entity in its configuration.
         Terrain::TerrainHeightGradientListConfig config;
-        config.m_gradientEntities.push_back(m_entity->GetId());
+        config.m_gradientEntities.push_back(entity->GetId());
 
-        Terrain::TerrainHeightGradientListComponent* heightGradientListComponent = m_entity->CreateComponent<Terrain::TerrainHeightGradientListComponent>(config);
+        auto heightGradientListComponent = entity->CreateComponent<Terrain::TerrainHeightGradientListComponent>(config);
         m_app.RegisterComponentDescriptor(heightGradientListComponent->CreateDescriptor());
 
+        return heightGradientListComponent;
+    }
+
+    void AddRequiredComponetsToEntity(AZ::Entity* entity)
+    {
+        // Create the required box component.
+        UnitTest::MockAxisAlignedBoxShapeComponent* boxComponent = entity->CreateComponent<UnitTest::MockAxisAlignedBoxShapeComponent>();
+        m_app.RegisterComponentDescriptor(boxComponent->CreateDescriptor());
+
         // Create a MockTerrainLayerSpawnerComponent to provide the required TerrainAreaService.
-        UnitTest::MockTerrainLayerSpawnerComponent* layerSpawner = m_entity->CreateComponent<UnitTest::MockTerrainLayerSpawnerComponent>();
+        UnitTest::MockTerrainLayerSpawnerComponent* layerSpawner = entity->CreateComponent<UnitTest::MockTerrainLayerSpawnerComponent>();
         m_app.RegisterComponentDescriptor(layerSpawner->CreateDescriptor());
-
-        m_entity->Init();
     }
 };
 
+TEST_F(TerrainHeightGradientListComponentTest, MissingRequiredComponentsActivateFailure)
+{
+    auto entity = CreateEntity();
+
+    AddHeightGradientListToEntity(entity.get());
+
+    const AZ::Entity::DependencySortOutcome sortOutcome = entity->EvaluateDependenciesGetDetails();
+    EXPECT_FALSE(sortOutcome.IsSuccess());
+
+    entity.reset();
```

Comment:
minor: now the `entity` is a local variable it will be automatically reset when the function finishes. All the `entity.reset();` calls in all the tests can be removed now.